lib/mix/lib/mix/tasks/deps.ex

```diff
@@ -120,6 +120,10 @@ defmodule Mix.Tasks.Deps do
     * `:subdir` - (since v1.13.0) search for the project in the given directory
       relative to the git checkout. This is similar to `:sparse` option but instead
       of a doing a sparse checkout it does a full checkout.
+    * `:depth` - (since v1.17.0) creates a shallow clone of the Git repository,
```

Comment:
This will require an update to the version depending on when this would be released.

Reply:
Great work @rhcarvalho! One question: do we want to expose the low-level concept of depth from git and allow arbitrary values to be passed to this option? I mean, If the idea is just to use `depth: 1` to improve performance, I think `shallow: true` could potentially represent it better. WDYT?

Reply:
I thought about it before and preferred `depth: 1` which maps directly to Git. If one knows Git, it becomes immediately clear what `mix deps.get` is supposed to do. If one doesn't know Git, `depth: 1` is no worse than `shallow: true`. The added benefit of sticking to Git is being able to support other flags that lead to a shallow clone, without ending up with ambiguous or duplicate options on Mix side.

Reply:
I think that in the end, it all depends on what we want to expose to users, but I personally, don't think mapping things one-to-one is the way to go (maybe because I see git internals as an implementation detail here). 

BTW, out of curiosity: do you know if we expose all git options for the `:sparse` key as well? (I can't find the supported options documented anywhere on hexdocs).

IMHO I think that `shallow: true` conveys the intention a lot better than `depth: 1`; **BUT** I think the question here is if we want to expose all git options or not... If not, the technique we use for shallow checkouts internally should remain an implementation detail and maybe hiding this behind a single configuration option is going to be better.

I'd love to get @josevalim's opinion on this.

Reply:
The main option is literally called :git, so staying closing to git is good. Especially because shallow in git means something else.

Reply:
AFAIK, this was always called a "shallow clone": https://git-scm.com/docs/shallow. But if you want to map one-to-one with git options this is probably the way. I do still think though that `shallow` is a better descriptor if we don't want to abstract a "performatic clone".

Reply:
You are right, I got the implementation details of shallow mixed up with its informal definition.

Reply:
This might not be relevant to the discussion, but I just wanted to point out that I think this **is** the formal definition of what shallow means in git though:

> To make a shallow clone, you can call "git-clone --depth 20 repo". The result contains only commit chains with a length of at most 20. It also writes an appropriate $GIT_DIR/shallow.

The docs for the `depth` also use the same terminology (which seems to be the case for the rest of the docs as well): https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt.

Reply:
Yes yes, no opposition on my side... Just wanted to clarify that the idea of "shallow cloning" is used consistently throughout the docs to describe the idea of fetching a limited portion of the commit history.

I thought the intent was just creating a more performant way of downloading deps (independently from the implementation), but it's clear that you actually want to map the git options directly. 

Great work nonetheless @rhcarvalho, cheers! 😊 👍